rocks-lib/src/operations/install.rs

```diff
@@ -20,6 +20,81 @@ use thiserror::Error;
 
 use super::{resolve::get_all_dependencies, SearchAndDownloadError};
 
+/// A rocks package installer, providing fine-grained control
+/// over how packages should be installed.
+/// Can install multiple packages in parallel.
+pub struct Install<'a> {
+    config: &'a Config,
+    package_db: Option<RemotePackageDB>,
+    packages: Vec<(BuildBehaviour, PackageReq)>,
+    pin: PinnedState,
+    progress: Option<Arc<Progress<MultiProgress>>>,
```

Comment:
nitpick: I just realized this exists in the other PRs (including the earlier PR that was merged) as well: we don't need to wrap this in an option, since the whole purpose of `Progress<T>` was to allow us to pass `NoProgress` :p

Reply:
The reason this is wrapped in an `Option` is so that we can fall back to default behaviour if the caller doesn't pass in one.
(right now, the default behaviour is to create a new `MultiProgress`).